astropy/io/fits/tests/test_header.py

```diff
@@ -562,6 +562,28 @@ def test_continue_card_with_equals_in_value(self):
                 '* 5.87359e-12 * MWAvg(Av=0.12)')
         assert c.comment == 'pysyn expression'
 
+    def test_final_continue_card_lacks_ampersand(self):
+        """
+        Regression test for https://github.com/astropy/astropy/issues/3282
+        """
+
+        h = fits.Header()
+        h['SVALUE'] = 'A' * 69
+        assert repr(h).splitlines()[-1] == _pad("CONTINUE  'AA'")
+
+    def test_final_continue_card_ampersand_removal_on_long_comments(self):
+        """
+        Regression test for https://github.com/astropy/astropy/issues/3282
+        """
+
+        c = fits.Card('TEST', 'long value' * 10, 'long comment &' * 10)
+        assert (str(c) ==
+            "TEST    = 'long valuelong valuelong valuelong valuelong valuelong valuelong &'  "
+            "CONTINUE  'valuelong valuelong valuelong value&'                                "
+            "CONTINUE  '&' / long comment &long comment &long comment &long comment &long    "
+            "CONTINUE  '&' / comment &long comment &long comment &long comment &long comment "
+            "CONTINUE  '' / &long comment &                                                  ")
```

Comment:
It would have been fine to make this a single test function (you can have more than one assert in a test case) but this is fine.